apps/openmw/mwworld/weather.cpp

```diff
@@ -1180,7 +1223,11 @@ inline void WeatherManager::calculateTransitionResult(const float factor, const
     mResult.mFogDepth = lerp(current.mFogDepth, other.mFogDepth, factor);
     mResult.mDLFogFactor = lerp(current.mDLFogFactor, other.mDLFogFactor, factor);
     mResult.mDLFogOffset = lerp(current.mDLFogOffset, other.mDLFogOffset, factor);
-    mResult.mWindSpeed = lerp(current.mWindSpeed, other.mWindSpeed, factor);
+
+    mResult.mCurrentWindSpeed = calculateWindSpeed(mCurrentWeather, mCurrentWindSpeed);
+    mResult.mNextWindSpeed = calculateWindSpeed(mNextWeather, mNextWindSpeed);
+
+    mResult.mWindSpeed = lerp(mResult.mCurrentWindSpeed, mResult.mNextWindSpeed, factor);
```

Comment:
I'm not really familiar with this part of the code, but for the fun of reviewing: Why is this not something like lerp(current.mCurrentWindSpeed, other.mNextWindSpeed)? All other member variables seem to follow the same formula, but this is different from all the others.

Reply:
I do not understand the question.
During weather transitions Morrowind simulates wind speed for both next weather and current weather. So we do a linear interpolation between simulated values. The `current` and `other` have only values, taken from INI file. They are not simulated values.

Reply:
In theory, it is possible to pass a previous wind speed to the `calculateResult` as argument, but it looks like a hack anyway.

Reply:
Ok. I guess that's pretty clear. I'm pretty much just being a duck here. https://en.wikipedia.org/wiki/Rubber_duck_debugging